provider/marathon/marathon_test.go

```diff
@@ -934,440 +699,563 @@ func TestMarathonGetProtocol(t *testing.T) {
 		c := c
 		t.Run(c.desc, func(t *testing.T) {
 			t.Parallel()
+			provider := &Provider{}
 			actual := provider.getProtocol(c.application)
 			if actual != c.expected {
-				t.Errorf("got protocol '%s', want '%s'", actual, c.expected)
+				t.Errorf("actual %q, expected %q", actual, c.expected)
 			}
 		})
 	}
 }
 
-func TestMarathonGetPassHostHeader(t *testing.T) {
-	provider := &Provider{}
-
-	applications := []struct {
+func TestMarathonGetSticky(t *testing.T) {
+	cases := []struct {
+		desc        string
 		application marathon.Application
 		expected    string
 	}{
 		{
-			application: marathon.Application{
-				Labels: &map[string]string{}},
+			desc:        "label missing",
+			application: application(),
```

Comment:
Okay, now I'm confused: The blog post uses _a/an_ and _with_ prefixes. However, our Docker builder comes with just two _with_ functions and otherwise uses no prefixes. ¯\_(ツ)_/¯

@vdemeester: can you help me out here? :)